agent/agent_test.go

```diff
@@ -42,3 +46,56 @@ func TestAgentListenMultipleClose(t *testing.T) {
 	Close()
 	Close()
 }
+
+func TestFormatBytes(t *testing.T) {
+	tests := []struct {
+		val      uint64
+		expected string
+	}{
+		{1023, "1023 bytes"},
+		{1024, "1KB (1024 bytes)"},
+		{1024*1024 - 1, "1023KB (1048575 bytes)"},
+		{1024 * 1024, "1MB (1048576 bytes)"},
+		{1024 * 1025, "1MB (1049600 bytes)"},
+		{1024 * 1024 * 1024 * 1024 * 1024, "1PB (1125899906842624 bytes)"},
+		{1024 * 1024 * 1024 * 1024 * 1024 * 1024, "1024PB (1152921504606846976 bytes)"},
+	}
+	for _, c := range tests {
+		result := formatBytes(c.val)
+		if result != c.expected {
+			t.Errorf("expected result %q for %v, got: %q", c.expected, c.val, result)
+		}
+	}
+}
+
+func TestHandleMemStats(t *testing.T) {
+	var buf bytes.Buffer
+	err := handle(&buf, []byte{signal.MemStats})
+	if err != nil {
+		t.Fatal(err)
+	}
+	expected := `alloc: \d+.*bytes.*
+total-alloc: \d+.*bytes.*
+sys: \d+.*bytes.*
+lookups: \d+
+mallocs: \d+
+frees: \d+
+heap-alloc: \d+.*bytes.*
+heap-sys: \d+.*bytes.*
+heap-idle: \d+.*bytes.*
+heap-in-use: \d+.*bytes.*
+heap-released: \d+.*bytes.*
+heap-objects: \d+
+stack-in-use: \d+.*bytes.*
+stack-sys: \d+.*bytes.*
+next-gc: when heap-alloc >= \d+.*bytes.*
+last-gc: -
```

Comment:
Are we interested to assert the formatting of the particular command? I think not. TestFormatBytes is sufficient.